integration-cli/docker_api_containers_test.go

```diff
@@ -256,28 +256,45 @@ func TestVolumesFromHasPriority(t *testing.T) {
 
 func TestGetContainerStats(t *testing.T) {
 	defer deleteAllContainers()
-	name := "statscontainer"
-
-	runCmd := exec.Command(dockerBinary, "run", "-d", "--name", name, "busybox", "top")
+	var (
+		name   = "statscontainer"
+		runCmd = exec.Command(dockerBinary, "run", "-d", "--name", name, "busybox", "top")
+	)
 	out, _, err := runCommandWithOutput(runCmd)
 	if err != nil {
 		t.Fatalf("Error on container creation: %v, output: %q", err, out)
 	}
+	type b struct {
+		body []byte
+		err  error
+	}
+	bc := make(chan b, 1)
 	go func() {
-		time.Sleep(4 * time.Second)
-		runCommand(exec.Command(dockerBinary, "kill", name))
-		runCommand(exec.Command(dockerBinary, "rm", name))
+		body, err := sockRequest("GET", "/containers/"+name+"/stats", nil)
+		bc <- b{body, err}
 	}()
 
-	body, err := sockRequest("GET", "/containers/"+name+"/stats", nil)
-	if err != nil {
-		t.Fatalf("GET containers/stats sockRequest failed: %v", err)
-	}
+	// allow some time to stream the stats from the container
+	time.Sleep(4 * time.Second)
+	runCommand(exec.Command(dockerBinary, "kill", name))
```

Comment:
Also, always better to check errors, because some future refactorer can change `run` for `rn` and didn't notice this.